packages/ooth-jwt/src/index.ts

```diff
@@ -15,22 +15,38 @@ type Options = {
   ooth: Ooth;
   tokenExpiry?: number;
   refreshTokenExpiry?: number;
-  sharedSecret: string;
+  sharedSecret?: string;
+  privateKey?: string;
+  publicKey?: string;
+  algorithm?: string;
 };
 
+type TokenOptions = {
+  sharedSecret?: string;
+  privateKey?: string;
+  publicKey?: string;
+  algorithm?: string;
+}
+
 function randomToken(): string {
   return randomBytes(43).toString('hex');
 }
 
-export function getToken(userId: string, iat: number, tokenExpiry: number, sharedSecret: string): string {
-  return sign(
-    {
-      iat,
-      exp: iat + tokenExpiry,
-      _id: userId,
-    },
-    sharedSecret,
-  );
+export function getToken(userId: string, iat: number, tokenExpiry: number, options: TokenOptions): string {
+  const data = {
+    iat,
+    exp: iat + tokenExpiry,
+    _id: userId,
+  };
+
+  if(options.sharedSecret) {
+    return sign(data, options.sharedSecret);
+  } else if(options.privateKey) {
+    return sign(data, options.privateKey, {
+      algorithm: options.algorithm
+    });
+  }
+  throw new Error('No secret nor key provided');
```

Comment:
Could you throw an error in the initialization function too?

Reply:
Done. I added 3 distinct errors:
- No secret nor key provided
- Both secret and key provided
- privateKey provided without publicKey